File: packages/server/app/classes/game/game-config.ts
import { PlayerData } from '@app/classes/communication/player-data';
import { RoundData } from '@app/classes/communication/round-data';
import Player from '@app/classes/player/player';
import { VirtualPlayerLevel } from '@app/classes/player/virtual-player-level';
import { Square } from '@app/classes/square';
import { TileReserveData } from '@app/classes/tile/tile.types';
import { Channel } from '@common/models/chat/channel';
import { WithIdOf } from '@common/types/id';

export interface GameConfigData {
    playerName: string;
    playerId: string;
    maxRoundTime: number;
    virtualPlayerLevel: VirtualPlayerLevel;
}

export interface GameConfig {
    player1: Player;
    maxRoundTime: number;
}

export interface ReadyGameConfig extends GameConfig {
    player2: Player;
    player3: Player;
    player4: Player;
}

export interface ReadyGameConfigWithChannelId extends ReadyGameConfig, WithIdOf<Channel> {}

export interface StartGameData {
    player1: PlayerData;
    player2: PlayerData;
    player3: PlayerData;
    player4: PlayerData;
    maxRoundTime: number;
    gameId: string;
    board: Square[][];
    tileReserve: TileReserveData[];
    round: RoundData;
}

File: packages/mobile/lib/classes/game/game-config.dart
import 'package:mobile/classes/game/player.dart';

import '../../components/tile/tile.dart';
import '../rounds/round.dart';

class StartGameData {
  final Player player1;
  final Player player2;
  final Player player3;
  final Player player4;
  final int maxRoundTime;
  final String gameId;
  final List<Tile> localPlayerTiles;
  final Round firstRound;

  StartGameData({
    required this.player1,
    required this.player2,
    required this.player3,
    required this.player4,
    required this.maxRoundTime,
    required this.gameId,
    required this.localPlayerTiles,
    required this.firstRound,
  });

  factory StartGameData.fromJson(Map<String, dynamic> json) {
    return StartGameData(player1: Player.fromJson(json['player1']), player2: Player.fromJson(json['player2']), player3: Player.fromJson(json['player3']), player4: Player.fromJson(json['player4']), maxRoundTime: json['maxRoundTime'], gameId: gameId, localPlayerTiles: localPlayerTiles, firstRound: firstRound)
  }
}
